don't speak unless given text, ignore > and <

Kebap opened the pull request:
#### Brief overview of PR changes/additions
only feed text to TTS engine when it contains more than whitespace, but remove > and < characters before

#### Motivation for adding to Mudlet
fix #4688
fix #4689

#### Other info (issues closed, discussion etc)
May be worthwile to raise this bug upsteam, but no idea how to build a minimal example without too much effort

Kebap committed:
don't speak when given whitespace

Kebap committed:
don't speak when given whitespace

Kebap committed:
don't speak when given whitespace

Kebap committed:
don't speak when given whitespace

Kebap commented:
Works for me, please review.

atari2600tim commented:
Works good on Windows 10, and code logic looks sensible enough.

github-actions[bot] committed:
(autocommit) Updated autocompletion data (#4709)

Co-authored-by: mudlet-machine-account <mudlet-machine-account@users.noreply.github.com>

vadi2 reviewed (approved):
Looks good, but I'd return nil+msg in case the text is autogenerated and someone wonders why isn't it doing anything

github-actions[bot] committed:
(autocommit) Updated text for translation (#4707)

Co-authored-by: mudlet-machine-account <mudlet-machine-account@users.noreply.github.com>

Kebap commented:
> return nil+msg

I was pondering that also.
In that case, need to await #4678 merged though.

evilmog committed:
CoreMUD listing addition (#4708)

vadi2 committed:
Make it clearer for Mudlet newbies where trigger text goes (#4673)

vadi2 committed:
Fix Mudlet Discord link to always work (#4697)

gcms committed:
Infrastructure: Fix the TMxpSendTagHandlerTest and re-enable it (#4714)

* Fixes problem blocking compilation of TMxpSendTagHandlerTest

- Extracts TMediaData to a header in order to remove dependencies on MediaPlayer in testing environment

* Adds override specifier to TMxpStubClient functions

SlySven committed:
Mapper: draw room exits that go off the map (#4716)

Retitled from:
"BugFix: correctly evaluate room m(in|ax)(x|y) values in all cases"

We previously set them all to be 0.0 when there is no custom lines but
technically, since they are supposed to be extremes of the room and it's
custom exit lines they should be initialised to the room's own coordinates.

Not doing this meant that in some cases they were being used but they
did not hold sensible values.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 committed:
Infrastructure: attempt to repair PTB changelog (#4721)

Kebap committed:
nil+err + replace angle brackets

Kebap committed:
nil+err + replace angle brackets

Kebap committed:
nil+err + replace angle brackets

Kebap committed:
nil+err + replace angle brackets

Kebap commented:
Enhanced this scope to include #4689 as well. Need to review again.

There is double code now, but did not refactor. 
My c fu wasn't strong enough.

SlySven committed:
BugFix: repair special exits in speedwalks with corrupt names (#4722)

This is an error that slipped in as a result of #4526 and resulted in the
exit room id being stored as the special exit name in the BGL graph that is
built for the A* search algorithm to work on. Consquently if that route was
used in a path the special exit command produced for that step was garbage
which also tripped off warnings about invalid characters being sent to the
MUD as an exit direction.

Thanks to @ryanstadther for bringing this to my attention!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap committed:
enable queue return to Lua

Kebap committed:
enable queue return to Lua

Kebap committed:
enable queue return to Lua

Kebap committed:
enable queue return to Lua

Kebap committed:
move queue instead

Kebap committed:
move queue instead

Kebap committed:
move queue instead

Kebap committed:
move queue instead

Kebap committed:
fix

Kebap committed:
fix

Kebap committed:
fix

Kebap committed:
fix

SlySven committed:
Don't crash when closing Mudlet (PTBs only) (#4723)

This should close #4682.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

gcms committed:
Fix bug on parsing MXP tags with UTF8 strings (#4717)

vadi2 committed:
Improve http error messages (#4725)

vadi2 committed:
Show value warnings in debug console (#4678)

Kebap committed:
Merge branch 'tts_dont_speak' of https://github.com/Mudlet/Mudlet into tts_dont_speak

Kebap commented:
Oh no, I push/pulled in a wrong way, now many recent commits from development (?) are shown here as well..

Kebap committed:
Merge branch 'tts_dont_speak' of https://github.com/Mudlet/Mudlet into tts_dont_speak

Kebap committed:
Use warnArgumentValue

Kebap committed:
Remove obsolete variable

Kebap committed:
Use warnArgumentValue more

Kebap commented:
Those last 3 commits were the actual ones intended. 
Ready for review.

Kebap committed:
Revert lots of unwanted commits.

atari2600tim commented:
fixes #3108
